azerothcore/azerothcore-wotlk
# fix(Core/SAI): Fix Kaw the Mammoth Destroyer quest

## Changes Proposed:
-  Fixed some smartAi events that were wrong
-  Made possible to finish the quest by making the Tauren attackable after killing the mount.
- Changed the class Unit.cpp to fix a bug that made the mammoth used as a mount uncontrollable after some seconds, forcing you to wander as a creature. Patch found here (https://github.com/TrinityCore/TrinityCore/issues/22368) and reburbished to work on AC.

Co-authored-by: Wyrserth <wyrserth@users.noreply.github.com>

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/12054

## SOURCE:

https://www.wowhead.com/quest=11879/kaw-the-mammoth-destroyer

## Tests Performed:

- .add quest 11879
- Check that you can mount a Wooly Mammoth Bull and that the skills work
- Check that after coming close to the creature the event starts, he yells and jump into the mammoth.
- Check that fter killing the mamoth you can kill him properly, and on death he drops the weapon to finish the quest.

## Known Issues and TODO List:


- Im using player near instead of the prefered areatrigger condition. Even though the areatrigger was reached i could not make him to do the event.
- When he is dead a yellow message and a warning sound should pop with the text "Kaw`s halberd breaks in two and falls to the ground" but i couldnt see how to make it on SAI.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
